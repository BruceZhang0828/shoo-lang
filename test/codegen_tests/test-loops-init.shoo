//add tests to declare any variable of any type in the for loop; and add failure tests to not allow anything except initializations and assignments


int i = 0;
// test to show any statement can be in for loop initializer
for(println("hello"); i<5; i++){
	println("bye");
}

println(str_of_int(i));
println("NICE");
int counter = 0;

// test to use other variables than int
for(bool x = true; x; println("how about this?")){
	if(counter > 3){
		x = false;
	}
	counter++;
	println(str_of_bool(x));
}
println("END");

// while loop testing
bool f = true;
while(f){
	println("stuff");
	f = false;
}

while(f){
	println("uh shouldnt print");
}