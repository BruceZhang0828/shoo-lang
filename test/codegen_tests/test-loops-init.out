hello
bye
bye
bye
bye
bye
5
NICE
true
how about this?
true
how about this?
true
how about this?
true
how about this?
false
how about this?
END
stuff